Replace listWeek with listMonth

felixhahnweilheim opened the pull request:
solves https://github.com/humhub/calendar/issues/463 by replacing listWeek with ~listYear~ listMonth.

@Semir1212 mentioned two options:
- replacing listWeek with ~listYear~ listMonth (what I did)
- adding listYear (would also be very simple to implement, we just not a wording to differentiate the two, like "List (Year)" and "List (Week)"

I think ideally this should be somehow configurable as there are networks with just a few events and those with a lot.

But I would like to bring this issue further because the listWeek view is not practical in my case.
Actually there would be also listMonth which could also be interesting.

With this PR I just want to show how easy this is to solve as the different view modes are already available.

luke- commented:
@felixhahnweilheim Thanks for the example and PR!

@Semir1212 @felixhahnweilheim  An extra config option for this is perhaps a bit much.
However, I am also unsure which view we should offer here.
Maybe we could find a dynamic solution for this? Depending on the number of entries, weekly/monthly/yearly?
Otherwise, I would opt for monthly as an in-between option.

felixhahnweilheim commented:
I updated the PR to use `listMonth`. I agree this is a good in between option.
I also updated that in the default view mode only `listMonth` is available (still called "List").
So the available default view modes are the same as the buttons shown in the FullCalendar view.

Old settings (listWeek, listYear) are handled in the `initSettings` method. This way we don't need a migration.

> Maybe we could find a dynamic solution for this? Depending on the number of entries, weekly/monthly/yearly?

This also sounds interesting. However users might not understand what is happening then. And it is a bit more effort to implement. So I think listMonth is a good solution. (Unless someone wants to spend the time code a good dynamic solution.)

felixhahnweilheim commented:
@luke- @Semir1212 Any news here?

I think the monthly list is a good solution now.

luke- commented:
@felixhahnweilheim Sorry for the delay. Sounds good with the Monthly list. Can you please add this change to the Changelog as well?

felixhahnweilheim commented:
Yes, done
